### PR TITLE
Adds case reporting support

### PR DESCRIPTION
**Pull request by kevgliss**
This PR duplicates functionality that we have with reporting incidents, allowing for the direct reporting of cases via a form. 

Screen Shot 2022-09-19 at 1 21 12 PM
Screen Shot 2022-09-19 at 1 21 40 PM

### CODE REVIEW

File: src/dispatch/static/dispatch/src/case/ReportSubmissionCard.vue [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      isSubmitted: false,
+      project_faq: null,
+    }
+  },
+
+  computed: {
+    ...mapFields("case_management", [
+      "selected.case_priority",
+      "selected.case_type",
+      "selected.commander",
```
Comment (mvilanova) [inline src/dispatch/static/dispatch/src/case/ReportSubmissionCard.vue:125]: Remove.

File: src/dispatch/static/dispatch/src/case/ReportSubmissionCard.vue [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      "selected.title",
+      "selected.tags",
+      "selected.description",
+      "selected.conversation",
+      "selected.conference",
```
Comment (mvilanova) [inline src/dispatch/static/dispatch/src/case/ReportSubmissionCard.vue:130]: Remove or comment out.

File: src/dispatch/static/dispatch/src/case/CaseSummaryTable.vue [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      hide-default-footer
+    >
+      <template v-slot:item.case_priority.name="{ item }">
+        <case-priority :priority="item.case_priority.name" />
+      </template>
+      <template v-slot:item.status="{ item }">
+        <case-status :status="item.status" :id="item.id" />
+      </template>
+      <template v-slot:item.reporter="{ item }">
+        <case-participant :participant="item.reporter" />
```
Comment (mvilanova) [inline src/dispatch/static/dispatch/src/case/CaseSummaryTable.vue:18]: There's no component for case participant. Did you forget to add it and import it?

**Review by mvilanova — CHANGES_REQUESTED** [review]

File: src/dispatch/static/dispatch/src/case/CaseSummaryTable.vue (hunk 3 above)
Comment (kevgliss) [reply src/dispatch/static/dispatch/src/case/CaseSummaryTable.vue:18]: I will remove it.

Comment (kevgliss) [reply src/dispatch/static/dispatch/src/case/CaseSummaryTable.vue:18]: Afaict it looks like cases don't track the reporter, I think that's probably something that should be added.

File: src/dispatch/static/dispatch/src/case/ReportReceiptCard.vue [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+  components: {},
+  data() {
+    return {
+      isSubmitted: false,
+      activeResourcePlugins: {
+        document: null,
+        ticket: null,
+        storage: null,
+        conversation: null,
+        conference: null,
```
[the remark this answers is not in the document]
Comment (kevgliss) [reply src/dispatch/static/dispatch/src/case/ReportReceiptCard.vue:180]: Is the question around whether a case would ever have an associated conference resource? I can't think of a reason, I will remove both and we can add them back when support is available.

File: src/dispatch/static/dispatch/src/case/CaseSummaryTable.vue (hunk 3 above)
Comment (mvilanova) [reply src/dispatch/static/dispatch/src/case/CaseSummaryTable.vue:18]: Yes, I agree we should add it.

File: src/dispatch/static/dispatch/src/case/ReportReceiptCard.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (mvilanova) [reply src/dispatch/static/dispatch/src/case/ReportReceiptCard.vue:180]: Sounds good.

**Review by mvilanova — APPROVED** [review]

**Pull request merged**